eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/actions/ActionExecutingOnClickListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.actions;
+
+import android.app.Activity;
+import android.view.View;
+
+
+/**
+ * A {@link View.OnClickListener} that executes the given {@link ActionExecutable} on onClick().
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class ActionExecutingOnClickListener implements View.OnClickListener
```

Comment:
Maybe call it just `ActionClickListener` everything else seems kind of redundant.